crane/app_util.py

```diff
@@ -199,3 +217,75 @@ def validate_and_transform_repoid(repo_id):
     if repo_id.startswith('library/'):
         return repo_id[len('library/'):]
     return repo_id
+
+
+def name_authorized(name):
+    response_data_v2 = get_data_v2()
+    repo_tuple_v2 = response_data_v2['repos'].get(name)
+
+    # if this deployment of this app does not know about the requested repo
+    if repo_tuple_v2 is None:
+        raise exceptions.HTTPError(httplib.NOT_FOUND)
+
+    if repo_tuple_v2.protected:
+        cert = _get_certificate()
+        if not cert or not cert.check_path(repo_tuple_v2.url_path):
+            # return 404 so we don't reveal the existence of repos that the user
+            # is not authorized for
+            raise exceptions.HTTPError(httplib.NOT_FOUND)
+
+
+def authorize_name(func):
```

Comment:
It's a little confusing that this function is called "authorize_name" and the one above it is called "name_authorized". I can't think of any specific name suggestions for either of these though. It might be OK to put the logic in "name_authorized" into "wrapper" below to avoid the confusion, though that would make authorize_name a little complex. I'm just rambling now.